spirv_glsl.cpp

```diff
@@ -1092,7 +1095,34 @@ void CompilerGLSL::emit_buffer_block_flattened(const SPIRVariable &var)
 	auto buffer_name = to_name(type.self, false);
 	size_t buffer_size = (get_declared_struct_size(type) + 15) / 16;
 
-	statement("uniform vec4 ", buffer_name, "[", buffer_size, "];");
+	SPIRType::BaseType basic_type;
+	if (get_common_basic_type(type, basic_type))
+	{
+		SPIRType tmp;
+		tmp.basetype = basic_type;
+
+		const char *flat_type = nullptr;
```

Comment:
This could use type_to_glsl(tmp) if tmp.vecsize is set to 4; we'll still need to check if the type is one of float/int/uint but that can be handled in one if(...) THROW().